### ISSUE TITLE
Input validation broken in v0.14: "Cannot use object of type Closure as array"

### ISSUE THREAD

**Issue by bravik** [overblog/GraphQLBundle#973]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Version/Branch   | 14.0.1

Getting Exception:
```
 Cannot use object of type Closure as array[0] (caught throwable) at /app/vendor/overblog/graphql-bundle/sr
c/Validator/InputValidator.php line 139
```
Here: 
https://github.com/overblog/GraphQLBundle/blob/a77a31483a725963892523b02960434a0a61c924/src/Validator/InputValidator.php#L139

Here are my definitions:
```yaml
FitnessSwapExerciseForProgramMutation:
  type: object
  config:
    fields:
      fitnessSwapExerciseForProgram:
        type: 'Void'
        args:
          input:
            type: 'FitnessSwapExerciseForProgramInput!'
            validation: cascade
        resolve: "@=mutation('fitness:program:swap_exercise', args['input'], getUser().getId())"
        access: "@=hasRole('ROLE_ATHLETE')"

FitnessSwapExerciseForProgramInput:
  type: input-object
  config:
    fields:
      exerciseId:
        type: 'ID!'
        validation:
          - NotBlank: ~
          - Type:
              type: numeric
      suitableLocation:
        type: 'FitnessExerciseLocation!'

FitnessExerciseLocation:
  type: enum
  config:
    description: Location type of exercises.
    values:
      ANY:
        value: 'any'
      GYM:
        value: 'gym'
      HOME:
        value: 'home'
```

Here is an input dump on the line before error:
```
InputValidator.php on line 138:
array:2 [
  "exerciseId" => array:2 [
    "type" => GraphQL\Type\Definition\NonNull {#4080
          // ...
      }
    }
    "validation" => Closure() {#4083
      class: "Overblog\GraphQLBundle\__DEFINITIONS__\FitnessSwapExerciseForProgramInputType"
      this: Overblog\GraphQLBundle\__DEFINITIONS__\FitnessSwapExerciseForProgramInputType {#3605 …}
    }
  ]
  "suitableLocation" => Closure() {#4022
    class: "Overblog\GraphQLBundle\__DEFINITIONS__\FitnessSwapExerciseForProgramInputType"
    this: Overblog\GraphQLBundle\__DEFINITIONS__\FitnessSwapExerciseForProgramInputType {#3605 …}
    use: {
      $services: Overblog\GraphQLBundle\Definition\GraphQLServices {#2145 …}
    }
  }
]
```

So the problem is with `suitableLocation` property, which is a closure.

I am not sure if it's not supposed to be a closure here,
or this is some regression introduced by recent `InputValidator` class refactoring in https://github.com/overblog/GraphQLBundle/pull/815
@murtukov 

I'm also having some similar errors witth validation after upgrading to v0.14:
* `Cannot use object of type GraphQL\Type\Definition\NonNull as array[0] `
* `Cannot use object of type GraphQL\Type\Definition\BooleanType as array[0] (caught throwable) at /app/vendo
r/overblog/graphql-bundle/src/Validator/InputValidator.php line 139.`
* `Array to string conversion[2] (caught throwable) at /app/vendor/symfony/validato
r/Constraints/Composite.php line 74`

**Comment by stevro** [comment]
Hi,
I think I encounter a similar problem after upgrading to 0.14.
The exception I encounter while running graphl:dump-schema command in TypeDecoratorListener.php line 171 is:
` Cannot use object of type GraphQL\Type\Definition\NonNull as array`

**Comment by murtukov** [comment]
@mcg-web wasn't it already fixed?

**Comment by SparkDragon** [comment]
Same errors for me after upgrading to 0.14

**Comment by bdunogier** [comment]
Same problem here on v0.14.1. Several of the fields that are being looped on are indeed not arrays but types.

**Comment by mcg-web** [comment]
Fixed and released in [v0.14.2](https://github.com/overblog/GraphQLBundle/releases/tag/v0.14.2)

**Comment by bdunogier** [comment]
Thank you !
